docs: Add Slack native review guide (#68221)
gravitational/teleport: docs/pages/identity-governance/access-requests/plugins/slack-review.mdx

  below your profile picture.
- Either a Linux host or Kubernetes cluster where you will run the Slack plugin.
- A Slack application, Bot OAuth token, and App-level token to use for the plugin.

(!/docs/pages/includes/tctl.mdx!)

## Step 1/7. Configure your Slack app

In this step, you will configure your Slack app to support the native Access Request reviews feature.

### Generate an app-level token

Enter the following URL and select your Slack app to reach your app's settings:
[`https://api.slack.com/apps`](https://api.slack.com/apps)

In your Slack app's settings, head to the **Basic Information** tab.

Then, under **App-Level Tokens**, click **Generate Token and Scopes**. Set a name
and add the `connections:write` scope. This will provide the plugin with the necessary permissions
to receive user interactions from Slack.

Copy the generated app-level token for use in a later step: <Var name="xapp-..." />

### Enable Socket Mode setting

Head to the **Socket Mode** tab and enable Socket Mode. This will allow the Teleport Slack plugin
to build a WebSocket connection with the Slack server in order to process Access Request reviews from Slack.

> ![Enable Slack Socket Mode](../../../../img/slack-socketmode.png)

Reviewer: ```suggestion
![Enable Slack Socket Mode](https://goteleport.com/_uploads/slack_socketmode_4badb28ce0.png)
```

<TabItem label="Docker">
```code
$ docker run -v <path-to-config>:/etc/teleport-slack.toml public.ecr.aws/gravitational/teleport-plugin-slack:(=teleport.version=) start
```
</TabItem>
<TabItem label="Helm Chart">
```code
$ helm upgrade --install teleport-plugin-slack teleport/teleport-plugin-slack --values teleport-slack-helm.yaml
```

To inspect the plugin's logs, use the following command:

```code
$ kubectl logs deploy/teleport-plugin-slack
```

Debug logs can be enabled by setting `log.severity` to `DEBUG` in
`teleport-slack-helm.yaml` and executing the `helm upgrade ...` command
above again. Then you can restart the plugin with the following command:

```code
$ kubectl rollout restart deployment teleport-plugin-slack
```
</TabItem>
</Tabs>

## Step 6/7. Test your Slack app

Once you've created the Slack app and the Teleport Slack plugin is
running, you can now test the workflow.

### Create an Access Request

(!docs/pages/includes/plugins/create-request.mdx!)

The user you configured earlier to review the request should receive a direct
message from "Teleport" in Slack with Access Request details.
This message will contain “Approve” and “Deny” buttons for the user to either

> approve or deny the request.

Author: outdated

> ![Slack Access Request message with approve and deny buttons](../../../../img/slack-review-request.png)

Reviewer: ```suggestion
![Slack Access Request message with approve and deny buttons](https://goteleport.com/_uploads/slack_review_request_20fcbbb8fe.png)
```

### Resolve the request

As the Slack user you set as an Access Request reviewer, click the "Approve" button,
and the plugin will send a reply to the original message with the review response.

Once the request is resolved, the Slack bot will add an emoji reaction of ✅ or
❌ to the Slack message for the Access Request, depending on whether the request
was approved or denied.

The Web UI will also reflect the resolved Access Request.

> ![Access Request reviewed on Web UI](../../../../img/slack-approve-request.png)

Reviewer: ```suggestion
![Access Request reviewed on Web UI](https://goteleport.com/_uploads/slack_approve_request_00532ec851.png)
```

The Teleport audit log will report the Access Request as submitted by the Teleport Slack plugin identity.
The `submitted_by` field represents the plugin identity that submitted on behalf of the `reviewer` user.

<Details title="Audit log">

> ![Audit log for Access Request review](../../../../img/slack-review-audit-log.png)

Reviewer: ```suggestion
![Audit log for Access Request review](https://goteleport.com/_uploads/slack_review_audit_log_628108813c.png)
```

---
title: Run the Slack Access Request Plugin with Native Access Request Reviews
description: How to set up Teleport's Slack plugin with native Access Request reviews.

> sidebar_label: Slack with Native Reviews

Reviewer 2: Maybe it's me but "native" seems a bit of a lower-level terminology. Can we rephrase these simpler along the lines of "Approving access requests from Slack" or something like that?

Author: I'm still unsure about the naming of the feature in the docs. If we don't mention "native review", we may come out to more verbose description eg. "Access Request reviews from Slack", which might also be vague/confusing.

I currently have the change without "native", but we may want to reconsider if "native" is the better alternative

tags:
 - access-requests
 - how-to
 - identity-governance
 - privileged-access
enterprise: Identity Governance
---

This guide will explain how to configure the Teleport Slack plugin to be able to review

> Access Requests natively within Slack.

Reviewer 2: ```suggestion
Access Requests within Slack.
```

  below your profile picture.
- Either a Linux host or Kubernetes cluster where you will run the Slack plugin.
- A Slack application, Bot OAuth token, and App-level token to use for the plugin.

(!/docs/pages/includes/tctl.mdx!)

## Step 1/7. Configure your Slack app

In this step, you will configure your Slack app to support the native Access Request reviews feature.

### Generate an app-level token

Enter the following URL and select your Slack app to reach your app's settings:
[`https://api.slack.com/apps`](https://api.slack.com/apps)

In your Slack app's settings, head to the **Basic Information** tab.

Then, under **App-Level Tokens**, click **Generate Token and Scopes**. Set a name
and add the `connections:write` scope. This will provide the plugin with the necessary permissions
to receive user interactions from Slack.

Copy the generated app-level token for use in a later step: <Var name="xapp-..." />

### Enable Socket Mode setting

Head to the **Socket Mode** tab and enable Socket Mode. This will allow the Teleport Slack plugin
to build a WebSocket connection with the Slack server in order to process Access Request reviews from Slack.

> ![Enable Slack Socket Mode](https://goteleport.com/_uploads/slack_socketmode_4badb28ce0.png)

Reviewer 2: You need to put all screenshots under [img](https://github.com/gravitational/teleport/tree/master/docs/img) folder and source them from there. See how other guides reference screenshots.

Author: Travis mentioned that we are transitioning image storage so I replaced the relative links with the Strapi links

Reviewer 2: Interesting, I didn't know that.

## Step 2/7. Define requester and reviewer users

Before you set up the Slack plugin, you will need to enable Role Access Requests
in your Teleport cluster.

For the purpose of this guide, we will define an `editor-requester` role that
can request the built-in `editor` role, and an `editor-reviewer` role that can

> review requests for the `editor` role.

Reviewer 2: For all these examples, let's use a less-privileged `access` role instead of `editor`.

```

Visit the invitation URL that is printed and login as `myuser` for the first time,
registering credentials as configured for your Teleport cluster.

Later in this guide, you will have `myuser` request the `editor` role so you can
review the request using the Teleport Slack plugin.

### Set up the reviewer user

Next, we will set up a user that can review Access Requests directly within Slack.
For this guide, we will bind the reviewer user based on Teleport username and Slack email.

Prepare a Slack user with an email address.
Create a Teleport user with a username set to the Slack user's email: <Var name="alice@example.com" />:

```code
$ tctl users add <Var name="alice@example.com" /> --roles=editor-reviewer
```

Visit the invitation URL that is printed and login as `<Var name="alice@example.com" />` for the first time,
registering credentials as configured for your Teleport cluster.

The Slack user will now be able to review Access Requests for the `editor` role within Slack
once we set up the Teleport Slack plugin.

For stronger identity binding, you can set up a Teleport trait
that contains the value of the Slack user ID, such as `slack_uid: U123456789`,

> but this will not be covered in this guide.

Reviewer 2: > this will not be covered in this guide

Why not? Feels like the guide should default to a more secure mode.

<Admonition type="info">
There are two identity binding models for Slack reviewers:

- binding based on Teleport username and Slack email
- binding based on Teleport trait and Slack user ID

The binding based on Teleport trait is stronger, but requires a Teleport user to be given a trait with

> their Slack user ID, often coming from the Identity Provider/SSO connector.

Reviewer 2: I think that SSO users case should generally be more upfront in this guide as it is what most people use, not local users. Maybe we can reformat this into a tabbed element for local/SSO users?

Also, we should explain how to set those traits (for local explain UI/commands), for SSO say that it should be configured in the IdP.

Author: I've added tabs for "Teleport trait binding" and "Teleport username binding." The "Teleport trait binding" also contains tabs for "Local users" & "SSO users". Also updated important details about local users in that section.

</Admonition>

> ## Step 3/7. Install the Teleport Slack plugin

Reviewer 2: Do we support configuring this for hosted Slack plugin in Cloud? If we do, we should explain how to do it, if we don't we should call it out explicitly somewhere (probably beginning of the guide).

Author: Not currently, will add a note near the beginning

```code
$ kubectl rollout restart deployment teleport-plugin-slack
```
</TabItem>
</Tabs>

## Step 6/7. Test your Slack app

Once you've created the Slack app and the Teleport Slack plugin is
running, you can now test the workflow.

### Create an Access Request

(!docs/pages/includes/plugins/create-request.mdx!)

The user you configured earlier to review the request should receive a direct
message from "Teleport" in Slack with Access Request details.
This message will contain “Approve” and “Deny” buttons for the user to either
approve or deny the request.

![Slack Access Request message with approve and deny buttons](https://goteleport.com/_uploads/slack_review_request_20fcbbb8fe.png)

### Resolve the request

As the Slack user you set as an Access Request reviewer, click the "Approve" button,
and the plugin will send a reply to the original message with the review response.

Once the request is resolved, the Slack bot will add an emoji reaction of ✅ or
❌ to the Slack message for the Access Request, depending on whether the request
was approved or denied.

The Web UI will also reflect the resolved Access Request.

> ![Access Request reviewed on Web UI](https://goteleport.com/_uploads/slack_approve_request_00532ec851.png)

Reviewer 2: Same here, update all images to point to local paths.

(and MFA for admin actions if enabled on cluster).
As a result, a compromised Slack account can approve requests from Slack and escalate privileges in Teleport.
</Admonition>

<Admonition type="info">
The Access Request reviews feature for the Teleport Slack plugin is currently
only for self-hosted plugins. Support for Teleport-hosted plugins will be added in the future.
</Admonition>

## How it works

Teleport's Slack integration notifies individuals and channels of
Access Requests. By enabling the plugin with Access Request reviews, users can then
approve or deny Access Requests directly within Slack without being redirected to the Teleport Web UI.

## Prerequisites

(!docs/pages/includes/edition-prereqs-tabs.mdx edition="Teleport Enterprise"!)

(!docs/pages/includes/machine-id/plugin-prerequisites.mdx!)

- Slack admin privileges to create an app and install it to your workspace. Your
  Slack profile must have the "Workspace Owner" or "Workspace Admin" banner
  below your profile picture.
- Either a Linux host or Kubernetes cluster where you will run the Slack plugin.
- A Slack application, Bot OAuth token, and App-level token to use for the plugin.

> 

Reviewer 3: Note: do not forget when backporting to v18 to specify the minimum version the feature is available in.

(!/docs/pages/includes/tctl.mdx!)

## Step 1/7. Configure your Slack app

In this step, you will configure your Slack app to support the Access Request reviews feature.

### Generate an app-level token

Enter the following URL and select your Slack app to reach your app's settings:
[`https://api.slack.com/apps`](https://api.slack.com/apps)

In your Slack app's settings, head to the **Basic Information** tab.

Then, under **App-Level Tokens**, click **Generate Token and Scopes**. Set a name
and add the `connections:write` scope. This will provide the plugin with the necessary permissions
to receive user interactions from Slack.

> Copy the generated app-level token for use in a later step: <Var name="xapp-..." />

Reviewer 3: Asking users to copy a secret app token into our website is very sketchy 😅

I don't know what type of telemetry we have, but regardless of its privacy property, we don't want to encourage risky behaviours.

Author: You are right 😅

### Enable Socket Mode setting

Head to the **Socket Mode** tab and enable Socket Mode. This will allow the Teleport Slack plugin

> to build a WebSocket connection with the Slack server in order to process Access Request reviews from Slack.

Author: Wrong, Socket Mode will by default turn it on.

<Admonition type="info">
All users performing Access Request reviews within Slack must have an associated local Teleport user.
When using an external Identity Provider, users should be imported using
the User Sync feature from the Teleport Okta integration or the Teleport Entra ID integration.
Teleport users that are SSO-only logins are short-lived and will expire,
requiring a re-login in Teleport to submit reviews in Slack.
</Admonition>

There are two identity binding models for Slack reviewers:
- binding based on Teleport username and Slack email
- binding based on Teleport trait and Slack user ID

The binding based on Teleport trait is stronger, but requires a Teleport user to be given a trait with
their Slack user ID, often coming from the Identity Provider/SSO connector.

You can set up both types of identity bindings if some users are imported
from an Identity Provider/SSO connector while others are local Teleport users.

<Tabs>
<TabItem label="Teleport trait binding">

Create a Teleport user with a username set as <Var name="alice@example.com" />
and assigned the role `access-reviewer`.

```code
$ tctl users add <Var name="alice@example.com" /> --roles=access-reviewer

> ```

Reviewer 3: This should be in the "local user" tab, SSO setups should not create users. Also TIL we don't have a `--trait` flag in `tctl users add`.

Visit the invitation URL that is printed and login as <Var name="alice@example.com" /> for the first time,
registering credentials as configured for your Teleport cluster.

We will assign the reviewer user with a Teleport trait named <Var name="slack_uid" />
that holds the value of the Slack user's ID: <Var name="U123456789" />.
The trait name can also be configured in the plugin configuration step.

<Tabs>
<TabItem label="Local users">

Create a file called `user-traits.yaml` with the following content:

```yaml
kind: user
version: v2
metadata:
  name: <Var name="alice@example.com" />
spec:
  roles: ['access-reviewer']
  traits:
    <Var name="slack_uid" />: ['<Var name="U123456789" />']
```

Update the newly created user for <Var name="alice@example.com" />:

```code
$ tctl create -f user-traits.yaml
```

</TabItem>
<TabItem label="SSO users">

When using a single sign-on user, the Teleport trait should be configured in the Identity Provider.

To get started integrating your Identity Provider with Teleport, read [Integrate
your Identity Provider](../../../zero-trust-access/sso/integrate-idp/integrate-idp.mdx).

> </TabItem>

Author: I'm not too familiar with setting up traits using an IdP; LMK if there is something else to add or a better link to redirect users.

```code
$ docker run -v <path-to-config>:/etc/teleport-slack.toml public.ecr.aws/gravitational/teleport-plugin-slack:(=teleport.version=) start
```
</TabItem>
<TabItem label="Helm Chart">
```code
$ helm upgrade --install teleport-plugin-slack teleport/teleport-plugin-slack --values teleport-slack-helm.yaml
```

To inspect the plugin's logs, use the following command:

```code
$ kubectl logs deploy/teleport-plugin-slack
```

Debug logs can be enabled by setting `log.severity` to `DEBUG` in
`teleport-slack-helm.yaml` and executing the `helm upgrade ...` command
above again. Then you can restart the plugin with the following command:

```code
$ kubectl rollout restart deployment teleport-plugin-slack
```
</TabItem>
</Tabs>

## Step 6/7. Test your Slack app

Once you've created the Slack app and the Teleport Slack plugin is
running, you can now test the workflow.

### Create an Access Request

(!docs/pages/includes/plugins/create-request.mdx role="access"!)

The user you configured earlier to review the request should receive a direct
message from "Teleport" in Slack with Access Request details.
This message will contain “Approve” and “Deny” buttons for the user to either

> approve or deny the request.

Author: outdated

![Slack Access Request message with approve and deny buttons](https://goteleport.com/_uploads/slack_review_request_20fcbbb8fe.png)

### Resolve the request

As the Slack user you set as an Access Request reviewer, click the "Approve" button,
and the plugin will send a reply to the original message with the review response.

Once the request is resolved, the Slack bot will add an emoji reaction of ✅ or
❌ to the Slack message for the Access Request, depending on whether the request
was approved or denied.

The Web UI will also reflect the resolved Access Request.

![Access Request reviewed on Web UI](https://goteleport.com/_uploads/slack_approve_request_00532ec851.png)

The Teleport audit log will report the Access Request as submitted by the Teleport Slack plugin identity.
The `submitted_by` field represents the plugin identity that submitted on behalf of the `reviewer` user.

<details>
<summary>Audit log</summary>

> ![Audit log for Access Request review](https://goteleport.com/_uploads/slack_review_audit_log_628108813c.png)

Reviewer 3: nit: put the JSON instead of a screenshot

---
title: Run the Slack Access Request Plugin with Access Request Reviews from Slack
description: How to set up Teleport's Slack plugin to review Access Requests within Slack.
sidebar_label: Slack with Access Request Reviews
tags:
 - access-requests
 - how-to
 - identity-governance
 - privileged-access
enterprise: Identity Governance
---

This guide will explain how to configure the Teleport Slack plugin to be able to review
Access Requests directly within Slack.

This guide assumes you are familiar with the [self-hosted Slack plugin guide](slack.mdx),
which covers initial plugin configuration and Slack app setup.

> <Admonition type="danger">

Reviewer 2: Nit: Maybe it's me but making this "danger" seems a bit harsh. I'd downgrade to "warning" but no strong opinion either way.

- A Slack application, Bot OAuth token, and App-level token to use for the plugin.

(!/docs/pages/includes/tctl.mdx!)

## Step 1/7. Configure your Slack app

In this step, you will configure your Slack app to support the Access Request reviews feature.

### Generate an app-level token

Enter the following URL and select your Slack app to reach your app's settings:
[`https://api.slack.com/apps`](https://api.slack.com/apps)

In your Slack app's settings, head to the **Basic Information** tab.

Then, under **App-Level Tokens**, click **Generate Token and Scopes**. Set a name
and add the `connections:write` scope. This will provide the plugin with the necessary permissions
to receive user interactions from Slack.

Copy the generated app-level token for use in a later step.

### Enable Socket Mode setting

Head to the **Socket Mode** tab and enable Socket Mode. This will allow the Teleport Slack plugin
to build a WebSocket connection with the Slack server in order to process Access Request reviews from Slack.

![Enable Slack Socket Mode](https://goteleport.com/_uploads/slack_socketmode_4badb28ce0.png)

## Step 2/7. Define requester and reviewer users

> Before you set up the Slack plugin, you will need to enable Role Access Requests

Reviewer 2: This works with resource access requests too, right? I imagine the notification/review doesn't really care about what kind of access request this is. Might want to mention it.

Author: I'm not really sure what this statement is meant to be saying, I'll remove it.

Create a file called `access-request-rbac.yaml` with the following content:

```yaml
kind: role
version: v8
metadata:
  name: access-reviewer
spec:
  allow:
    review_requests:
      roles: ['access']
---
kind: role
version: v8
metadata:
  name: access-requester
spec:
  allow:
    request:
      roles: ['access']
      thresholds:
        - approve: 1
          deny: 1
```

Create the roles you defined:

```code
$ tctl create -f access-request-rbac.yaml
role 'access-reviewer' has been created
role 'access-requester' has been created
```

(!docs/pages/includes/create-role-using-web.mdx!)

### Set up the requester user

Create a user called `myuser` who has the `access-requester` role:

```code
$ tctl users add myuser --roles=access-requester
```

Visit the invitation URL that is printed and login as `myuser` for the first time,
registering credentials as configured for your Teleport cluster.

Later in this guide, you will have `myuser` request the `access` role so you can
review the request using the Teleport Slack plugin.

### Set up the reviewer user

Next, we will set up a user that can review Access Requests directly within Slack.

<Admonition type="info">

> All users performing Access Request reviews within Slack must have an associated local Teleport user.

Reviewer 2: "local" user here is misleading, it usually implies user created with "tctl users create" which is rarely used in production (outside some maybe admin users).

I'd rephrase to "persistent" instead of local.

Reviewer 2: Also, this should be in the requirements as well.

When using an external Identity Provider, users should be imported using

> the User Sync feature from the Teleport Okta integration or the Teleport Entra ID integration.

Reviewer 2: Link to respective guides?

Teleport users that are SSO-only logins are short-lived and will expire,

> requiring a re-login in Teleport to submit reviews in Slack.

Reviewer 2: I'd rephrase a bit to make it clear that SSO users will be able to submit reviews for the duration of their ephemeral SSO user existing in Teleport and will need to re-login when it expires.

your Identity Provider](../../../zero-trust-access/sso/integrate-idp/integrate-idp.mdx).

</TabItem>
</Tabs>

</TabItem>
<TabItem label="Teleport username binding">

Prepare a Slack user with an email address.
Create a Teleport user with a username set to the Slack user's email <Var name="alice@example.com" />
and assigned the role `access-reviewer`.

```code
$ tctl users add <Var name="alice@example.com" /> --roles=access-reviewer
```

Visit the invitation URL that is printed and login as <Var name="alice@example.com" /> for the first time,
registering credentials as configured for your Teleport cluster.

</TabItem>
</Tabs>

The Slack user will now be able to review Access Requests for the `access` role within Slack
once we set up the Teleport Slack plugin.

## Step 3/7. Install the Teleport Slack plugin

(!docs/pages/includes/plugins/install-access-request.mdx name="slack"!)

## Step 4/7. Set up the Teleport Slack plugin

In this section, you will set up the Teleport Slack plugin 
and generate credentials that the plugin will use for authentication.

### Enable issuing of credentials for the plugin user

> The required permissions for the plugin are configured in the preset `access-plugin-with-review`

Reviewer 2: Add this role to the list of preset roles?

https://goteleport.com/docs/reference/access-controls/roles/#preset-roles

Author: Added in #67948

---

> title: Run the Slack Access Request Plugin with Access Request Reviews from Slack

Reviewer 4: If we're going to have two separate Slack guides, we need to make sure that each points to the other in its introduction, making it clear for users when to choose each guide. Right now, it's not clear from the sidebar and introduction of the main Slack guide which documentation a user should follow.

<img width="957" height="881" alt="Image" src="https://github.com/user-attachments/assets/5c100f33-c99e-4259-9182-3f73018bb13a" />

> title: Run the Slack Access Request Plugin with Access Request Reviews from Slack

Reviewer 4: This title is difficult to read in on the rendered docs page. Is there a more elegant, concise title we can think of?

<img width="639" height="378" alt="Image" src="https://github.com/user-attachments/assets/f0257325-0284-4cac-b90c-0d3dc580a7a3" />

Author: In that case I think "Run the Slack Access Request Plugin with Native Reviews" would be more concise. Any other suggestions are welcome though.

description: How to set up Teleport's Slack plugin to review Access Requests within Slack.
sidebar_label: Slack with Access Request Reviews
tags:
 - access-requests
 - how-to
 - identity-governance
 - privileged-access
enterprise: Identity Governance
---

This guide will explain how to configure the Teleport Slack plugin to be able to review
Access Requests directly within Slack.

This guide assumes you are familiar with the [self-hosted Slack plugin guide](slack.mdx),
which covers initial plugin configuration and Slack app setup.

<Admonition type="warning">
Enabling this feature reduces the security of the Teleport cluster.
Access Request reviews submitted through Slack are bound to a Slack identity rather than
the reviewer's Teleport identity, and bypass Teleport authentication
(and MFA for admin actions if enabled on cluster).
As a result, a compromised Slack account can approve requests from Slack and escalate privileges in Teleport.
</Admonition>

> <Admonition type="info">

Reviewer 4: Consecutive Admonitions make this difficult to read. Can we move any of this information to the Prerequisites?

<img width="627" height="528" alt="Image" src="https://github.com/user-attachments/assets/2d78d665-2329-4c22-9875-9a7d15edcd7d" />

Author: Sure, I can move the info admonition into the intro paragraph. I think this will be clearer to users that only the self-hosted plugin supports the feature instead of being hidden in the prereqs

<summary>Suggested reviewers</summary>

Users can suggest reviewers when they create an Access Request, e.g.:

```code
$ tsh request create --roles=dbadmin --reviewers=alice@example.com,ivan@example.com
```

If an Access Request includes suggested reviewers, the Slack plugin will add
these to the list of channels to notify. If a suggested reviewer is an email
address, the plugin will look up the direct message channel for that
address and post a message in that channel.

</details>

### Invite the plugin to your Slack app

Once you have configured the channels that the Slack plugin will notify when it
receives an Access Request, you will need to ensure that the plugin can post in
those channels.

You have already configured the plugin to send direct messages as Slackbot. For
any other channel you mention in your `role_to_recipients` map, you will need
to invite the plugin to that channel. Navigate to each channel and enter `/invite
@teleport` in the message box.

### Start the plugin

Start the plugin by following the instructions below.

<Tabs>
<TabItem label="Executable">
```code
$ teleport-slack start --config teleport-slack.toml
```
</TabItem>
<TabItem label="Docker">
```code

> $ docker run -v teleport-slack.toml:/etc/teleport-slack.toml public.ecr.aws/gravitational/teleport-plugin-slack:(=teleport.version=) start

Author: I noticed a discrepancy in docker setup in general, the identity file is not mounted in the docker container for the plugin to use. In addition, using machine identity requires a different path to mount. May need to consider extra tabs for docker setup: "Machine ID" & "Long-lived identity".

       The result should look like this:

       ![API Scopes](../../../../img/enterprise/plugins/slack/api-scopes.png)

     1. After you have configured scopes for your plugin, scroll back to the top
        of the OAuth & Permissions page, find the "OAuth Tokens for Your
        Workspace" section, and click "Install to Workspace". You will see a
        summary of the permission you configured for the Slack plugin earlier.

     1. In "Where should Teleport post?", choose "Slackbot" as the default
        channel the plugin will post to. The plugin will post here when sending
        direct messages.  Later in this guide, we will configure the plugin to
        post in other channels as well.

        After submitting this form, you will see an OAuth token in the "OAuth &
        Permissions" tab under "Tokens for Your Workspace":

        ![OAuth Tokens](../../../../img/enterprise/plugins/slack/OAuth.png)

        You will use this token later when configuring the Slack plugin.

  </details>

(!docs/pages/includes/tctl.mdx!)

## Step 1/7. Configure your Slack app

In this step, you will configure your Slack app to support the native Access Request reviews feature.

> ### Generate an app-level token

Reviewer 3: Can you add an admonition reminding that the App token is different from the Bot token? If the user doesn't know this subtlety they will think they already have a Slack token.
